Fail
file path is :
projects/deltachat-core/rust/src/location.rs

output is:


error is :
   Compiling deltachat v1.139.6 (/home/k/ogs_PHD/LLM4SE/2Rust_projects/auto_get_functionPair/projects/deltachat-core/rust)
error[E0252]: the name `time` is defined multiple times
  --> src/location.rs:33:37
   |
17 | use crate::tools::time;
   |     ------------------ previous import of the value `time` here
...
33 | use crate::tools::{duration_to_str, time};
   |                                     ^^^^ `time` reimported here
   |
   = note: `time` must be defined only once in the value namespace of this module

error[E0425]: cannot find value `DC_CONTEXT_MAGIC` in this scope
   --> src/location.rs:354:52
    |
354 |     if context.inner.is_null() || context.magic != DC_CONTEXT_MAGIC || (latitude == 0.0 && longitude == 0.0) {
    |                                                    ^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `chat_id` in this scope
   --> src/location.rs:363:61
    |
363 |             &[latitude, longitude, accuracy, time() as i64, chat_id, DC_CONTACT_ID_SELF],
    |                                                             ^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_CONTACT_ID_SELF` in this scope
   --> src/location.rs:363:70
    |
363 |             &[latitude, longitude, accuracy, time() as i64, chat_id, DC_CONTACT_ID_SELF],
    |                                                                      ^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_CONTACT_ID_SELF` in this scope
   --> src/location.rs:367:44
    |
367 |         context.emit_location_changed(Some(DC_CONTACT_ID_SELF)).await?;
    |                                            ^^^^^^^^^^^^^^^^^^ not found in this scope

warning: unused import: `std::sync::Arc`
  --> src/location.rs:14:5
   |
14 | use std::sync::Arc;
   |     ^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> src/lib.rs:6:5
   |
6  |     unused,
   |     ^^^^^^
   = note: `#[warn(unused_imports)]` implied by `#[warn(unused)]`

warning: unused import: `std::time::SystemTime`
  --> src/location.rs:15:5
   |
15 | use std::time::SystemTime;
   |     ^^^^^^^^^^^^^^^^^^^^^

warning: unused imports: `Mutex`, `RwLock`
  --> src/location.rs:16:19
   |
16 | use tokio::sync::{Mutex, RwLock};
   |                   ^^^^^  ^^^^^^

warning: unused import: `time`
  --> src/location.rs:33:37
   |
33 | use crate::tools::{duration_to_str, time};
   |                                     ^^^^

error[E0599]: no method named `is_null` found for struct `Arc<InnerContext>` in the current scope
   --> src/location.rs:354:22
    |
354 |     if context.inner.is_null() || context.magic != DC_CONTEXT_MAGIC || (latitude == 0.0 && longitude == 0.0) {
    |                      ^^^^^^^ method not found in `Arc<InnerContext>`

error[E0609]: no field `magic` on type `&context::Context`
   --> src/location.rs:354:43
    |
354 |     if context.inner.is_null() || context.magic != DC_CONTEXT_MAGIC || (latitude == 0.0 && longitude == 0.0) {
    |                                           ^^^^^ unknown field
    |
    = note: available field is: `inner`
    = note: available fields are: `blobdir`, `sql`, `smeared_timestamp`, `generating_key_mutex`, `oauth2_mutex` ... and 17 others

error[E0308]: mismatched types
   --> src/location.rs:356:19
    |
356 |         return Ok(continue_streaming);
    |                -- ^^^^^^^^^^^^^^^^^^ expected `bool`, found integer
    |                |
    |                arguments to this enum variant are incorrect
    |
help: the type constructed contains `{integer}` due to the type of the argument passed
   --> src/location.rs:356:16
    |
356 |         return Ok(continue_streaming);
    |                ^^^------------------^
    |                   |
    |                   this argument influences the type of `Ok`
note: tuple variant defined here
   --> /home/k/.rustup/toolchains/1.77.1-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs:506:5
    |
506 |     Ok(#[stable(feature = "rust1", since = "1.0.0")] T),
    |     ^^

error[E0599]: no method named `execute` found for reference `&context::Context` in the current scope
   --> src/location.rs:359:30
    |
359 |     let stmt_chats = context.execute("SELECT id FROM chats WHERE locations_send_until>?;", &[time() as i64]).await?;
    |                              ^^^^^^^ method not found in `&Context`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following traits define an item `execute`, perhaps you need to implement one of them:
            candidate #1: `hyper::rt::Executor`
            candidate #2: `hyper::rt::Executor`
help: one of the expressions' fields has a method of the same name
    |
359 |     let stmt_chats = context.inner.sql.execute("SELECT id FROM chats WHERE locations_send_until>?;", &[time() as i64]).await?;
    |                              ++++++++++
help: one of the expressions' fields has a method of the same name
    |
359 |     let stmt_chats = context.sql.execute("SELECT id FROM chats WHERE locations_send_until>?;", &[time() as i64]).await?;
    |                              ++++

error[E0599]: no method named `execute` found for reference `&context::Context` in the current scope
   --> src/location.rs:361:35
    |
361 |         let stmt_insert = context.execute(
    |                           --------^^^^^^^ method not found in `&Context`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following traits define an item `execute`, perhaps you need to implement one of them:
            candidate #1: `hyper::rt::Executor`
            candidate #2: `hyper::rt::Executor`
help: one of the expressions' fields has a method of the same name
    |
361 |         let stmt_insert = context.inner.sql.execute(
    |                                   ++++++++++
help: one of the expressions' fields has a method of the same name
    |
361 |         let stmt_insert = context.sql.execute(
    |                                   ++++

error[E0308]: mismatched types
   --> src/location.rs:363:46
    |
363 |             &[latitude, longitude, accuracy, time() as i64, chat_id, DC_CONTACT_ID_SELF],
    |                                              ^^^^^^^^^^^^^ expected `f64`, found `i64`
    |
help: you can convert an `i64` to an `f64`, producing the floating point representation of the integer, rounded if necessary
    |
363 |             &[latitude, longitude, accuracy, (time() as i64) as f64, chat_id, DC_CONTACT_ID_SELF],
    |                                              +             ++++++++

error[E0599]: no method named `interrupt_location` found for reference `&context::Context` in the current scope
   --> src/location.rs:368:17
    |
368 |         context.interrupt_location().await;
    |                 ^^^^^^^^^^^^^^^^^^ method not found in `&Context`
    |
help: one of the expressions' fields has a method of the same name
    |
368 |         context.inner.scheduler.interrupt_location().await;
    |                 ++++++++++++++++
help: one of the expressions' fields has a method of the same name
    |
368 |         context.scheduler.interrupt_location().await;
    |                 ++++++++++

error[E0308]: mismatched types
   --> src/location.rs:371:8
    |
371 |     Ok(continue_streaming)
    |     -- ^^^^^^^^^^^^^^^^^^ expected `bool`, found integer
    |     |
    |     arguments to this enum variant are incorrect
    |
help: the type constructed contains `{integer}` due to the type of the argument passed
   --> src/location.rs:371:5
    |
371 |     Ok(continue_streaming)
    |     ^^^------------------^
    |        |
    |        this argument influences the type of `Ok`
note: tuple variant defined here
   --> /home/k/.rustup/toolchains/1.77.1-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs:506:5
    |
506 |     Ok(#[stable(feature = "rust1", since = "1.0.0")] T),
    |     ^^

Some errors have detailed explanations: E0252, E0308, E0425, E0599, E0609.
For more information about an error, try `rustc --explain E0252`.
warning: `deltachat` (lib) generated 4 warnings
error: could not compile `deltachat` (lib) due to 13 previous errors; 4 warnings emitted
warning: build failed, waiting for other jobs to finish...
warning: `deltachat` (lib test) generated 4 warnings (4 duplicates)
error: could not compile `deltachat` (lib test) due to 13 previous errors; 4 warnings emitted
error: command `/home/k/.rustup/toolchains/1.77.1-x86_64-unknown-linux-gnu/bin/cargo test --no-run --message-format json-render-diagnostics` exited with code 101

